src/main/java/codeu/controller/ChatServlet.java

```diff
@@ -76,6 +77,27 @@ void setUserStore(UserStore userStore) {
     this.userStore = userStore;
   }
 
+  /**
+   * This function formats any URL's within a message into HTML links.
+   */
+  String formatURL(String messageContent){
+    // URL regex
+    String patternString = "(http://www\\.|https://www\\.|http://|https://)+[a-z0-9]+([\\-\\.]{1}[a-z0-9]+)*\\.[a-z]{2,5}(:[0-9]{1,5})?(/.*)?";
```

Comment:
This is very clever!

Reply:
I can't take credit for it! I found the pattern online.